### ISSUE TITLE
Invalid value in tutorial

### ISSUE THREAD

**Issue by abedra** [GaloisInc/llvm-verifier#6]
When running `make check` from the code folder in the tutorial, I get the following result rather than `Valid` like the tutorial shows:

```
Loading LLVM implementation
Bitblasting Cryptol implementation
Checking equivalence
Invalid: 115662822337806967864241073576085482245194316165030070846998742492792053295122
```

**Comment by abedra** [comment]
More info:

```
abc 01> cec ./aes.aig ./aes-ref.aig 
Networks are NOT EQUIVALENT.
Verification failed for at least 64 outputs:  po000 po005 po007 ...
Output po000: Value in Network1 = 0. Value in Network2 = 1.
Input pattern:  pi000=0 pi128=0 pi001=1 pi129=0 pi002=0 pi130=0 pi003=0 pi131=0 pi004=1 pi132=0 pi005=0 pi133=0 pi006=0 pi134=0 pi007=0 pi135=0 pi040=0 pi168=0 pi041=0 pi169=0 pi042=0 pi170=0 pi043=0 pi171=0 pi044=0 pi172=0 pi045=0 pi173=0 pi046=0 pi174=0 pi047=0 pi175=0 pi080=0 pi208=0 pi081=0 pi209=0 pi082=0 pi210=0 pi083=0 pi211=0 pi084=0 pi212=0 pi085=0 pi213=0 pi086=0 pi214=0 pi087=0 pi215=0 pi120=0 pi248=0 pi121=0 pi249=0 pi122=0 pi250=0 pi123=0 pi251=0 pi124=0 pi252=0 pi125=0 pi253=0 pi126=0 pi254=0 pi127=0 pi255=0 pi232=0 pi233=0 pi234=0 pi235=0 pi236=0 pi237=0 pi238=0 pi239=0 pi032=0 pi160=0 pi033=0 pi161=0 pi034=0 pi162=0 pi035=0 pi163=0 pi036=0 pi164=0 pi037=0 pi165=0 pi038=0 pi166=0 pi039=0 pi167=0 pi072=0 pi200=0 pi073=0 pi201=0 pi074=0 pi202=0 pi075=0 pi203=0 pi076=0 pi204=0 pi077=0 pi205=0 pi078=0 pi206=0 pi079=0 pi207=0 pi112=0 pi240=0 pi113=0 pi241=0 pi114=0 pi242=0 pi115=0 pi243=0 pi116=0 pi244=0 pi117=0 pi245=0 pi118=0 pi246=0 pi119=0 pi247=0 pi024=0 pi152=0 pi025=0 pi153=0 pi026=0 pi154=0 pi027=0 pi155=0 pi028=0 pi156=0 pi029=0 pi157=0 pi030=0 pi158=0 pi031=0 pi159=0 pi136=0 pi137=0 pi138=0 pi139=0 pi140=0 pi141=0 pi142=0 pi143=0 pi064=0 pi192=0 pi065=0 pi193=0 pi066=0 pi194=0 pi067=0 pi195=0 pi068=0 pi196=0 pi069=0 pi197=0 pi070=0 pi198=0 pi071=0 pi199=0 pi104=0 pi105=0 pi106=0 pi107=0 pi108=0 pi109=0 pi110=0 pi111=0 pi016=0 pi144=0 pi017=0 pi145=0 pi018=0 pi146=0 pi019=0 pi147=0 pi020=0 pi148=0 pi021=0 pi149=0 pi022=0 pi150=0 pi023=0 pi151=0 pi056=0 pi184=0 pi057=0 pi185=0 pi058=0 pi186=0 pi059=0 pi187=0 pi060=0 pi188=0 pi061=0 pi189=0 pi062=0 pi190=0 pi063=0 pi191=0 pi176=0 pi177=0 pi178=0 pi179=0 pi180=0 pi181=0 pi182=0 pi183=0 pi096=0 pi224=0 pi097=0 pi225=0 pi098=0 pi226=0 pi099=0 pi227=0 pi100=0 pi228=0 pi101=0 pi229=0 pi102=0 pi230=0 pi103=0 pi231=0 pi008=0 pi009=0 pi010=0 pi011=0 pi012=0 pi013=0 pi014=0 pi015=0 pi048=0 pi049=0 pi050=0 pi051=0 pi052=0 pi053=0 pi054=0 pi055=0 pi088=0 pi216=0 pi089=0 pi217=0 pi090=0 pi218=0 pi091=0 pi219=0 pi092=0 pi220=0 pi093=0 pi221=0 pi094=0 pi222=0 pi095=0 pi223=0
```

**Comment by abedra** [comment]
And one more look:

```
sawscript> f <- read_aig "aes.aig";
sawscript> import "AES.cry";
Loading module AES
[warning] at ./AES.cry:156:1--159:39:
  Defaulting type parameter 'bits'
             of finite enumeration
             at ./AES.cry:156:36--156:51
  to 4
[warning] at ./AES.cry:156:1--159:39:
  Defaulting type parameter 'bits'
             of literal or demoted expression
             at ./AES.cry:156:60--156:63
  to 4
[warning] at ./AES.cry:133:1--138:31:
  Defaulting type parameter 'bits'
             of literal or demoted expression
             at ./AES.cry:136:32--136:33
  to 3
[warning] at ./AES.cry:100:1--102:23:
  Defaulting type parameter 'bits'
             of finite enumeration
             at ./AES.cry:101:61--101:69
  to 2
[warning] at ./AES.cry:95:1--97:20:
  Defaulting type parameter 'bits'
             of finite enumeration
             at ./AES.cry:96:58--96:66
  to 2
sawscript> let {{ aesExtract x = aesEncrypt (pt,key) where [pt,key] = split (join (reverse (groupBy`{32} x))) }};
sawscript> res <- prove abc {{ \x -> f x == aesExtract x }};
sawscript> print res
Invalid: 72370055773322622139731865630429942408293740416025352524660990004945706024960
```

**Comment by abedra** [comment]
I setup a different machine today with a different OS (original request was Fedora 22, new data is OS X) and got the

**Comment by atomb** [comment]
It turns out that the bit ordering in AIGs generated by LSS when using the SAW backend changed recently (to make it more consistent with the rest of the system). This means that the byte reversal that this script previously did is no longer necessary. I've committed a fix (9ca398e).

**Comment by abedra** [comment]
:+1:
